src/server/auth/server/http_client.go

```diff
@@ -0,0 +1,20 @@
+package server
+
+import (
+	"net/http"
+)
+
+type RewriteRoundTripper struct {
+	RewriteHost string
```

Comment:
TODO: This should support rewriting the protocol (`http` versus `https`) and it should check the URL to be rewritten.